processor/agentmetricsprocessor/utils_combine_process_metrics.go

```diff
@@ -0,0 +1,274 @@
+// +-------------------+ +-----------------------+     +-----------------------+
+// |  Resource: Empty  | |  Resource: Process 1  |     |  Resource: Process X  |
+// +-------+---+-------+ +---------+---+---------+ ... +---------+---+---------+
+// |Metric1|...|MetricN| |MetricN+1|...|MetricN+M|     |MetricN+1|...|MetricN+M|
+// +-------+---+-------+ +---------+---+---------+     +---------+---+---------+
+//
+// Converted format:
+//
+//                             ResourceMetrics
+// +---------------------------------------------------------------------+
+// |                           Resource: Empty                           |
+// +-------+---+-------+----------------------+---+----------------------+
+// |Metric1|...|MetricN|MetricN+1{P1, ..., PX}|...|MetricN+M{P1, ..., PX}|
+// +-------+---+-------+----------------------+---+----------------------+
+//
+// Assumptions:
+// * There is at most one resource metrics slice without process resource info (will raise error if not)
+// * There is no other resource info supplied that needs to be retained (info may be silently lost if it exists)
+// * All metrics with the same name have the same descriptor (process metrics will be merged if the names match)
+
+func combineProcessMetrics(rms pdata.ResourceMetricsSlice) error {
+	// create collection of combined process metrics, disregarding any ResourceMetrics
+	// with no process resource attributes as "otherMetrics"
+	processMetrics, otherMetrics, err := createProcessMetrics(rms)
+	if err != nil {
+		return err
+	}
+
+	// if no non-process metrics were supplied, initialize an empty
+	// ResourceMetrics object
+	if otherMetrics.IsNil() {
+		otherMetrics.InitEmpty()
+		ilms := otherMetrics.InstrumentationLibraryMetrics()
+		ilms.Resize(1)
+		ilms.At(0).InitEmpty()
+	}
+
+	// append the process metrics to the other ResourceMetrics object
+	metrics := otherMetrics.InstrumentationLibraryMetrics().At(0).Metrics()
+	// ideally, we would Resize & Set, but function not available at this time
+	for _, metric := range processMetrics {
+		metrics.Append(&metric.Metric) // also, why does append take a pointer?
+	}
+
+	// TODO: This is super inefficient. Instead, we should just return a new
+	// data.MetricData struct, but currently blocked as it is internal
+	rms.Resize(1)
+	otherMetrics.CopyTo(rms.At(0))
+	return nil
+}
+
+func createProcessMetrics(rms pdata.ResourceMetricsSlice) (processMetrics convertedMetrics, otherMetrics pdata.ResourceMetrics, err error) {
+	processMetrics = convertedMetrics{}
+	otherMetrics = pdata.NewResourceMetrics()
+
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		resource := rm.Resource()
+
+		// if these ResourceMetrics do not contain process resource attributes,
+		// these must be the "other" non-process metrics
+		if !includesProcessAttributes(resource) {
+			if !otherMetrics.IsNil() {
+				err = errors.New("unexpectedly received multiple Resource Metrics  with err")
```

Comment:
What does "with err" mean, sorry?
I think it should say something like "without process attributes" or similar

Reply:
Your comment is exactly right thanks! Must have been a typo